features/admin.users.v1/components/user-profile.tsx

```diff
@@ -2636,13 +2661,187 @@ export const UserProfile: FunctionComponent<UserProfilePropsInterface> = (
+     *
+     * Is true if the "Set Password" option should be presented—this indicates that the
+     * account is in the pending ask password state (i.e. the user hasn’t set a password yet).
+     *
+     * Else false if the "Force Password Reset" option should be displayed, meaning the user
+     * already has an existing password.
+     */
+    const isSetPassword: boolean =
+        (accountState === AccountState.PENDING_AP) ||
+        (accountLockedReason === AccountLockedReason.PENDING_ASK_PASSWORD);
+
+    /**
+     * Determines whether the "Resend" link should be displayed.
+     *
+     * The resend option is shown when:
+     * - The account is locked due to  "PENDING_ASK_PASSWORD",
+     *   indicating an initial password setup link has been sent.
+     * - The account is locked due to"PENDING_ADMIN_FORCED_USER_PASSWORD_RESET",
+     *   indicating that an admin-forced password reset has been sent.
+     * - The account is not locked but account state is "PENDING_AP",
+     *   indicating that the user hasn't set their password via the setup link yet.
+     *
+     * @returns whether the "Resend" link option should be displayed.
+     */
+    const showResendLink: boolean =
+        accountLockedReason === AccountLockedReason.PENDING_ASK_PASSWORD ||
+        accountLockedReason === AccountLockedReason.PENDING_ADMIN_FORCED_USER_PASSWORD_RESET ||
+        accountState === AccountState.PENDING_AP;
+
+    /**
+     * Initiates a recovery process based on the account's locked reason or account state.
+     *
+     **/
+    const handleResendCode = async () => {
+        setIsSubmitting(true);
+
+        const recoveryScenario: string | null = resolveRecoveryScenario();
+
+        if (!recoveryScenario) {
+            onAlertFired({
+                description: t("user:profile.notifications.resendCode.inValidRecoveryScenarioError.description"),
+                level: AlertLevels.ERROR,
+                message: t("user:profile.notifications.resendCode.inValidRecoveryScenarioError.message")
+            });
+        }
+
+        try {
+            const requestData: ResendCodeRequest = {
+                properties: [
+                    {
+                        key: "RecoveryScenario",
+                        value: recoveryScenario
+                    }
+                ],
+                user: {
+                    realm: user[ SCIMConfigs.scim.systemSchema ]?.userSource,
+                    username: user?.userName
+                }
+            };
+
+            await resendCode(requestData);
+            onAlertFired({
+                description: t("user:profile.notifications.resendCode.success.description",
+                    { recoveryOption: RECOVERY_SCENARIO_TO_RECOVERY_OPTION_TYPE_MAP[recoveryScenario] }),
+                level: AlertLevels.SUCCESS,
+                message: t("user:profile.notifications.resendCode.success.message")
+            });
+        } catch (error: any) {
```

Comment:
Shall we use proper type here rather than `any`?

Reply:
Resolved via https://github.com/wso2/identity-apps/pull/7992/commits/070b89ffe8036eac1f8ea66388f17d6f1226f175